File: src/main/java/com/lannstark/lec07/Lec07Main.java
package com.lannstark.lec07;

import java.io.IOException;

public class Lec07Main {

  private int parseIntOrThrow(String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      throw new IllegalArgumentException(String.format("%s 는 숫자가 아닙니다", str));
    }
  }

  private Integer parseIntOrThrow2(String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      return null;
    }
  }

  public static void main(String[] args) throws IOException {
    JavaFilePrinter javaFilePrinter = new JavaFilePrinter();
    javaFilePrinter.readFile("src/main/java/com/lannstark/lec07/a.txt");
  }

}

File: src/main/java/com/lannstark/lec07/Lec07Main.kt
package com.lannstark.lec07

fun parseIntOrThrow(str: String): Int {
    //kotlin에서는 try catch가 expression이다
    return try {
        str.toInt() //기본 타입간의 형변환은 toType을 사용한다
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("$str 는 숫자가 아닙니다")
    }
}

fun parseINtOrThrow2(str: String): Int? {
    return try {
        str.toInt()
    } catch (e: NumberFormatException) {
        null
    }
}

fun main() {
    FilePrinter().readFile("src/main/java/com/lannstark/lec07/a.txt")
}
